compiler/rustc_session/src/session.rs

```diff
@@ -1411,6 +1414,13 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
         }
     }
 
+    // LLVM MemTagSanitizer requires '+mte' target feature.
+    if sess.is_sanitizer_memtag_enabled() {
+        if !sess.opts.cg.target_feature.contains("+mte") {
```

Comment:
This will not work as intended if the user specifies `-Ctarget-feature=+mte,-mte`. The correct approach would be to call `llvm_util::global_llvm_features`, but it won't be available to a session here, so you would need to move this check to codegen time and fail there.

(An alternative option could be to enable the `mte` feature implicitly, though I'm sure that solution has its own tradeoffs…)

Reply:
Done, thanks!